TanukiKit/Projects.swift

```diff
@@ -0,0 +1,230 @@
+    open var starCount: Int?
+    open var forksCount: Int?
+    open var openIssuesCount: Int?
+    open var runnersToken: String?
+    open var publicBuilds: Bool?
+    open var createdAt: Date?
+    open var lastActivityAt: Date?
+    open var lfsEnabled: Bool?
+    open var visibilityLevel: Int?
+//    open var sharedWithGroups: [SharedWithGroups]?
+    open var onlyAllowMergeIfBuildSucceeds: Bool?
+    open var requestAccessEnabled: Bool?
+    open var permissions: Permissions?
+
+    public init(_ json: [String: AnyObject]) {
+        owner = User(json["owner"] as? [String: AnyObject] ?? [:])
+        if let id = json["id"] as? Int {
+            self.id = id
+            name = json["name"] as? String
+            nameWithNamespace = json["path_with_namespace"] as? String
+            isPublic = json["public"] as? Bool
+            projectDescription = json["description"] as? String
+            if let urlString = json["ssh_url_to_repo"] as? String, let url = URL(string: urlString) { sshURL = url }
+            if let urlString = json["http_url_to_repo"] as? String, let url = URL(string: urlString) { cloneURL = url }
+            if let urlString = json["web_url"] as? String, let url = URL(string: urlString) { webURL = url }
+            path = json["path"] as? String
+            pathWithNamespace = json["path_with_namespace"] as? String
+            contaierRegisteryEnabled = json["container_registry_enabled"] as? Bool
+            defaultBranch = json["default_branch"] as? String
+            tagList = json["tag_list"] as? [String]
+            isArchived = json["archived"] as? Bool
+            issuesEnabled = json["issues_enabled"] as? Bool
+            mergeRequestsEnabled = json["merge_requests_enabled"] as? Bool
+            wikiEnabled = json["wiki_enabled"] as? Bool
+            buildsEnabled = json["builds_enabled"] as? Bool
+            snippetsEnabled = json["snippets_enabled"] as? Bool
+            sharedRunnersEnabled = json["shared_runners_enabled"] as? Bool
+            publicBuilds = json["public_builds"] as? Bool
+            creatorID = json["creator_id"] as? Int
+            namespace = Namespace(json["namespace"] as? [String: AnyObject] ?? [:])
+            if let urlString = json["avatar_url"] as? String, let url = URL(string: urlString) { avatarURL = url }
+            starCount = json["star_count"] as? Int
+            forksCount = json["forks_count"] as? Int
+            openIssuesCount = json["open_issues_count"] as? Int
+            visibilityLevel = json["visibility_level"] as? Int
+            createdAt = Time.rfc3339Date(string: json["created_at"] as? String)
+            lastActivityAt = Time.rfc3339Date(string: json["last_activity_at"] as? String)
+            lfsEnabled = json["lfs_enabled"] as? Bool
+            runnersToken = json["runners_token"] as? String
+//            sharedWithGroups = SharedWithGroups(json["shared_with_groups"] as? [String: AnyObject] ?? [:]) Define as list of objects?
```

Comment:
How would I declare this as a list? AS you can see from the updated fixtures, they can contain many objects of `SharedWithGroups`.